src/bundle/Resources/public/js/CKEditor/anchor/anchor-editing.js

```diff
@@ -17,6 +17,57 @@ class IbexaAnchorEditing extends Plugin {
                 key: 'id',
             },
         });
+
+        this.editor.conversion.for('dataDowncast').add((dispatcher) => {
+            dispatcher.on('attribute:anchor:listItem', (event, data, conversionApi) => {
+                if (data.attributeKey !== 'anchor' || data.attributeNewValue === '') {
+                    return;
+                }
+
+                const viewItem = conversionApi.mapper.toViewElement(data.item);
+                const previousElement = viewItem.parent.previousSibling;
+
+                conversionApi.writer.setAttribute('id', data.attributeNewValue, viewItem.parent);
+                conversionApi.writer.removeAttribute('id', viewItem);
+
+                if (previousElement?.name === viewItem.parent.name) {
```

Comment:
Maybe we could assign `if`s conditions to variables to make them easier to understand?